src/Shared/RequestDataHelper.cs

```diff
@@ -78,7 +81,27 @@ public void SetActivityDisplayName(Activity activity, string originalHttpMethod,
         var normalizedHttpMethod = this.GetNormalizedHttpMethod(originalHttpMethod);
         var namePrefix = normalizedHttpMethod == "_OTHER" ? "HTTP" : normalizedHttpMethod;
 
-        activity.DisplayName = string.IsNullOrEmpty(httpRoute) ? namePrefix : $"{namePrefix} {httpRoute}";
+        activity.DisplayName = GetDisplayName();
+
+        string GetDisplayName()
+        {
+            if (string.IsNullOrEmpty(httpRoute))
+            {
+                return namePrefix;
+            }
+
+            if (DisplayNameCache.TryGetValue(httpRoute!, out var displayName))
+            {
+                return displayName;
+            }
+
+            if (DisplayNameCache.Count < DisplayNameCacheSize)
```

Comment:
`.Count` gets a lock on all buckets, so this won't be efficient on the hot path

Reply:
Yes that's what the benchmark shows above.

Reply:
Yes, that's what the benchmark above shows.